priorityclass/priorityclass_test.go

```diff
@@ -0,0 +1,135 @@
+package priorityclass
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func createDefaultPriorityClasses(c *fake.Clientset) {
+	c.SchedulingV1().PriorityClasses().Create(context.Background(), &schedulingv1.PriorityClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system-cluster-critical",
+		},
+		Value: 2000000000,
+	}, metav1.CreateOptions{})
+	c.SchedulingV1().PriorityClasses().Create(context.Background(), &schedulingv1.PriorityClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system-node-critical",
+		},
+		Value: 2000001000,
+	}, metav1.CreateOptions{})
+}
+
+var listOfDefaultPriorityClases = []schedulingv1.PriorityClass{
+	{
+		ObjectMeta: metav1.ObjectMeta{Name: "system-cluster-critical"},
+		Value:      2000000000,
+	},
+	{
+		ObjectMeta: metav1.ObjectMeta{Name: "system-node-critical"},
+		Value:      2000001000,
+	},
+}
+
+func TestService_DeleteCollection(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name                   string
+		prepareFakeClientSetFn func() *fake.Clientset
+		lopt                   metav1.ListOptions
+		wantErr                bool
+		wantReturn             *schedulingv1.PriorityClassList
+	}{
+		{
+			name: "delete all priorityclasses",
```

Comment:
@sanposhiho I attempted to do some testing here, but I believe I cannot actually delete collections with the fake client https://github.com/kubernetes/client-go/issues/609. This test is failing because I still get the deleted priorityclasses after the deletecollection:
```
              	Items: []v1.PriorityClass{
            - 		{ObjectMeta: v1.ObjectMeta{Name: "development"}, Value: 600000},
            - 		{ObjectMeta: v1.ObjectMeta{Name: "production"}, Value: 1000000},
            - 		{ObjectMeta: v1.ObjectMeta{Name: "staging"}, Value: 800000},
              		{ObjectMeta: {Name: "system-cluster-critical"}, Value: 2000000000},
              		{ObjectMeta: {Name: "system-node-critical"}, Value: 2000001000},
              	},
```

Reply:
You can use Reactor to check the behavior that isn't supported in fake-client.
https://github.com/kubernetes/client-go/blob/671a40c59fcad6cf1d31592e79dab1effa1768ae/testing/fake.go#L98